### PR TITLE
Formspec: Add support for vertical scrollbar in textarea

### PR DESCRIPTION
**Pull request by adelcoding1**
- textarea Formspec uses now a `GUIEditBoxWithScrollBar` instead of the default `irr::gui::IGUIEditBox`
- `GUIEditBoxWithScrollBar` allows to use a vertical scrollbar when it's needed
- `GUIEditBoxWithScrollBar` implements `irr::gui::IGUIEditBox`

### CODE REVIEW

File: build/android/jni/Android.mk [hunk 1]
```diff
@@ -149,6 +149,7 @@ LOCAL_SRC_FILES := \
 		jni/src/genericobject.cpp                 \
 		jni/src/gettext.cpp                       \
 		jni/src/guiChatConsole.cpp                \
+		jni/src/guieditboxwithscrollbar.cpp       \
```
Comment (nerzhul) [inline build/android/jni/Android.mk]: please use same filename as other gui*.cpp files

File: src/guieditboxwithscrollbar.h [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+	u32 m_blink_start_time;
+	s32 m_cursor_pos;
+	s32 m_hscroll_pos, m_vscroll_pos; // scroll position in characters
+	u32 m_max;
+
+	bool m_word_wrap, m_multiline, m_autoscroll, m_passwordbox;
+	wchar_t m_passwordchar;
+	EGUI_ALIGNMENT m_halign, m_valign;
+
+	std::vector< core::stringw > m_broken_text;
```
Comment (nerzhul) [inline src/guieditboxwithscrollbar.h]: no space in template types

File: src/guieditboxwithscrollbar.cpp [hunk 3]
```diff
@@ -0,0 +1,1484 @@
...
+
+//! called if an event happened.
+bool GUIEditBoxWithScrollBar::OnEvent(const SEvent& event)
+{
+	if (isEnabled()){
+		switch (event.EventType)
+		{
+		case EET_GUI_EVENT:
+			if (event.GUIEvent.EventType == EGET_ELEMENT_FOCUS_LOST)
+			{
```
Comment (nerzhul) [inline src/guieditboxwithscrollbar.cpp]: please fix the code style and place your braces on upper lines for conditions, switch & for

File: src/guieditboxwithscrollbar.cpp [hunk 4]
```diff
@@ -0,0 +1,1484 @@
...
+	if (!event.KeyInput.PressedDown)
+		return false;
+
+	bool text_changed = false;
+	s32 new_mark_begin = m_mark_begin;
+	s32 new_mark_end = m_mark_end;
+
+	// control shortcut handling
+
+	if (event.KeyInput.Control)	{
```
Comment (nerzhul) [inline src/guieditboxwithscrollbar.cpp]: missing space

**Review by nerzhul — CHANGES_REQUESTED** [review]

File: src/guiFormSpecMenu.cpp [hunk 5]
```diff
@@ -1153,8 +1154,8 @@ void GUIFormSpecMenu::parseTextArea(parserData* data,
 		} else {
 #else
 		{
-#endif
-			e = Environment->addEditBox(spec.fdefault.c_str(), rect, true, this, spec.fid);
+#endif			
+			e = new GUIEditBoxWithScrollBar(spec.fdefault.c_str(), true, Environment,this, spec.fid, rect, 20);
```
Comment (SmallJoker) [inline src/guiFormSpecMenu.cpp]: Tailing indents after `#endif`
Missing space after `true, Environment`

**Comment by Ekdohibs** [comment]
Since this is a modification of irrlicht code, maybe it would be better to put it in the corresponding folder.

**Comment by nerzhul** [comment]
Seems your merged instead of rebase, please do git rebase master

**Comment by kilbith** [comment]
New PR here : https://github.com/minetest/minetest/pull/5355

**Pull request closed without merging**